lib/utils/types.ts

```diff
@@ -2,6 +2,22 @@ import * as t from "io-ts";
 
 import { Set as SerializableSet } from "json-set-map";
 
+/**
+ * Returns a subset of the input objects fields.
+ *
+ * @param obj the input object
+ * @param props an array of keys to preserve
+ */
+export const pick = <T, K extends keyof T>(
```

Comment:
I don't like this method as it loses type information, in general I'm very much against using `Object#*` methods and `Array#reduce` as they basically discard all the type information about an object - in fact you can see it from the signature of the method, if the method returns only a subset of keys, ho can the output type be the same type of `obj`?

Reply:
which obj ? the type of the object that pick returns is responsibility of the caller. ie:

```
f(): SomeType {
  pick(...)   // <--- MUST return SomeType
}

f() {
  const a : SomeType = pick(...)   // <--- MUST return SomeType
}
```

Reply:
ok :)